EIPS/eip-4910.md

```diff
@@ -0,0 +1,1160 @@
+    2. For an ERC20 payment token, the Buyer must then call the `executePayment` in the NFT contract -- the ERC20 is not directly involved.
+* For a non-ERC20 payment, the Buyer must send a protocol token (ETH) to the NFT contract, and is required to send `msg.data` encoded as an array of purchased NFTs `uint256[] tokenId`.
+
+#### Modified NFT Transfer Function including required Trade data to allocate Royalties
+
+The input parameters must satisfy several requirements for the NFT to be transferred AFTER the royalties have been properly distributed. Furthermore, the ability to transfer more than one token at a time is also considered.
+
+The proposal defines:
+* Input parameter validation
+* Payment Parameter Validation
+* Distributing Royalties
+* Update Royalty Account ownership with payout
+* Transferring Ownership of the NFT 
+* Removing the Payment entry in `registeredPayment` after successful transfer
+
+##### Distributing Royalties
+
+The approach to distributing royalties is to break down the hierarchical structure of interconnected Royalty Accounts into layers and then process one layer at time, where each relationship between a token and its ancestor is utilized to traverse the Royalty Account chain until the root ancestor and associated RA is reached.
+
+#### Paying out Royalties to the NFT Owner -- `from` address in `safeTransferFrom` Function
+
+This is the final part of the proposal.
+
+There are two versions of the payout function -- a `public` function and an `internal` function.
+
+The public function has the following interface:
+```
+function royaltyPayOut (uint256 tokenId, address RAsubaccount, address payable payoutAccount, uint256 amount) public virtual nonReentrant returns (bool)
+```
+
+where we only need the `tokenId`, the RA Sub Account address, `_RAsubaccount` which is the `owner`, and the amount to be paid out, `_amount`. Note that the function has `nonReentrant` modifier protection, because funds are being payed out.
+
+#### Sending a Payout Payment
+
+The following steps need to be taken:
+* find the RA Sub Account based on `RAaccount` and the `subaccountPos` and extract the balance
+* extract `tokentype` from the Sub Account
+* based on the token type, send the payout payment (not exceeding the available balance)
+
+### Data Structures
+
+#### Rationale: Royalty Account and Royalty Sub Accounts
+
+In order to create an interconnected data structure linking NFTs to RAs that is search optimized requires to make the following additions to the global data structures of an ERC721.
+
+Note, a Royalty Account is defined as a collection of Royalty Sub Accounts linked to a meta account. This meta account is comprised of general account identifiers particular to the NFT it is linked to such as asset identifier, parent identifier etc.
+
+#### Specification: Royalty Account and Royalty Sub Accounts
+
+##### **[R1]**
```

Comment:
I haven't tried it myself, but would footnotes (`[^1]`) work for these? Failing that, maybe `<a id="r1">...</a>` is supported?

Reply:
will give it a whirl and see if it works @SamWilsn

Reply:
done ... using <a name="name_tag">...</a> works